### ISSUE TITLE
utf8 case folding and/or comparisons?

### ISSUE THREAD

**Issue by liquidaty** [simdutf/simdutf#135]
Hi,

Thank you for your work on this. Would it be possible to use this (or any other SIMD-based) approach for fast case folding and/or case-insensitive comparison?

**Comment by WojciechMula** [comment]
At https://github.com/SnellerInc/sneller we have Unicode support and even for upper/lower-case we ended up with huge lookup tables. Here we might consider using gathers, and check the outcome. Although I'm quite sure performance will be mainly bounded by cache misses penalties.

**Comment by lin72h** [comment]
@WojciechMula I didn't know you are one of the sneller developer, very impressive project BTW!

**Comment by WojciechMula** [comment]
> @WojciechMula I didn't know you are one of the sneller developer, very impressive project BTW!

Yeah, a lot of great stuff is there. :) Getting back to the main topic, there was an attempt to express lookup table as a huge vectorized if-ladder and it was very slow.
